src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,196 @@
+            )
+            self._add_refresh_token(refresh_token)
+            refresh_token_with_id = self._get_refresh_token_by_token(refresh_token)
+
+            access_token = create_access_token(
+                payload, client, self.env_config.secret_key
+            )
+            self._add_access_token(refresh_token_with_id.id, access_token)
+
+            return {"access_token": access_token, "refresh_token": refresh_token}
+
+    def login_user(self, client_id, client_secret, user_name, password):
+        client = self._get_client_by_client_id(client_id)
+        user = self._get_user_by_user_name(user_name)
+
+        if bcrypt.checkpw(
+            client_secret.encode("utf-8"), client.client_secret.encode("utf-8")
+        ) and bcrypt.checkpw(password.encode("utf-8"), user.password.encode("utf-8")):
+
+            payload = {
+                "user_name": user_name,
+                "name": user.name,
+                "last_name": user.last_name,
+                "role": user.role,
+                "client_id": client.id,
+            }
+
+            refresh_token = create_refresh_token(
+                payload,
+                client,
+                self.env_config.secret_key,
+            )
+            self._add_refresh_token(refresh_token)
+            refresh_token_with_id = self._get_refresh_token_by_token(refresh_token)
+
+            access_token = create_access_token(
+                payload,
+                client,
+                self.env_config.secret_key,
+            )
+            self._add_access_token(refresh_token_with_id.id, access_token)
+
+            return {"access_token": access_token, "refresh_token": refresh_token}
+
+    def validate_token(self, token):
+        try:
+            access_token_decoded = jwt.decode(
+                token, self.env_config.secret_key, algorithms="HS256"
+            )
+
+            return True
+
+        except jwt.ExpiredSignatureError:
+            self._delete_access_token(token)
+            return False
+        except jwt.InvalidSignatureError:
+            return False
+        except jwt.DecodeError:
+            return False
+
+    def refresh_token(self, refresh_token):
+        try:
+            refresh_token_in_db = self._get_refresh_token_by_token(refresh_token)
+
+            refresh_token_decoded = jwt.decode(
+                refresh_token, self.env_config.secret_key, algorithms="HS256"
+            )
+
+            client = self._get_client_by_id(refresh_token_decoded["client_id"])
+
+            new_access_token = create_new_access_token(
+                refresh_token_decoded, client, self.env_config.secret_key
+            )
+
+            self._add_access_token(refresh_token_in_db.id, new_access_token)
+
+            return new_access_token
+
+        except jwt.ExpiredSignatureError:
+            self._delete_refresh_token(refresh_token)
+
+    def _get_user_by_user_name(self, user_name):
+        with self.engine.begin() as conn:
```

Comment:
Please use the userservice if you have one.

Reply:
I don't have an user services for the moment. Raw query will be replace when user services will created.